### PR TITLE
Test OSS-Fuzz with introspector images [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by Navidem**

### CODE REVIEW

**Comment by Navidem** [comment]
/gcbrun trial_build.py all --sanitizer address --fuzzing-engine libfuzzer

**Comment by Navidem** [comment]
/gcbrun trial_build.py all --sanitizer address coverage memory undefined --fuzzing-engine libfuzzer afl honggfuzz centipede

**Comment by Navidem** [comment]
/gcbrun trial_build.py all --sanitizer coverage memory undefined --fuzzing-engine afl honggfuzz centipede

**Comment by jonathanmetzman** [comment]
What happened to this? why were these failing?

**Comment by Navidem** [comment]
/gcbrun trial_build.py all --sanitizer coverage --fuzzing-engine libfuzzer

**Comment by Navidem** [comment]
> /gcbrun trial_build.py all --sanitizer coverage memory undefined --fuzzing-engine afl honggfuzz centipede



> What happened to this? why were these failing?

7 projects failed because of timeout: `Step #11: Error response from daemon: Get "https://gcr.io/v2/": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)`

Individual failures:
`snappy` is already failing to build for the main branch.
`libtorrent` is failing with this message: `2022-09-29 17:03:52 ERROR 503: Egress is over the account limit..`
`cryptofuzz` failing with `abort: error: No route to host`
 
I am going to try different configurations one by one.

**Comment by Navidem** [comment]
> /gcbrun trial_build.py all --sanitizer coverage --fuzzing-engine libfuzzer

This timed out!

**Comment by Navidem** [comment]
/gcbrun trial_build.py htslib --sanitizer coverage --fuzzing-engine libfuzzer

**Comment by Navidem** [comment]
/gcbrun trial_build.py airflow arrow bloaty boost curl ffmpeg openssh servo tor wasm3 wamstime zlib --sanitizer coverage --fuzzing-engine libfuzzer

**Comment by Navidem** [comment]
/gcbrun trial_build.py airflow arrow bloaty boost curl ffmpeg openssh servo tor wasm3 wamstime zlib --sanitizer memory undefined --fuzzing-engine libfuzzer

**Comment by Navidem** [comment]
/gcbrun trial_build.py all --sanitizer address --fuzzing-engine afl honggfuzz centipede

**Comment by jonathanmetzman** [comment]
> > What happened to this? why were these failing?
> 
> 7 projects failed because of timeout: `Step #11: Error response from daemon: Get "https://gcr.io/v2/": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)`
> 
> Individual failures: `snappy` is already failing to build for the main branch. `libtorrent` is failing with this message: `2022-09-29 17:03:52 ERROR 503: Egress is over the account limit..` `cryptofuzz` failing with `abort: error: No route to host`
> 
> I am going to try different configurations one by one.

These seem like flake and not the fault of your change.

**Comment by Navidem** [comment]
This trial build fails:
```
trial_build.py all --sanitizer address --fuzzing-engine afl honggfuzz centipede
```

Some errors I could spot:
`gstreamer`:
```
Step #19 - "compile-afl-address-x86_64": ../src/gstreamer/meson.build:1:0: ERROR: Meson version is 0.61.3 but project requires >= 0.62.0
Step #19 - "compile-afl-address-x86_64": 
Step #19 - "compile-afl-address-x86_64": A full log can be found at /work/_builddir/meson-logs/meson-log.txt
Step #19 - "compile-afl-address-x86_64": ********************************************************************************
Step #19 - "compile-afl-address-x86_64": Failed to build.
Step #19 - "compile-afl-address-x86_64": To reproduce, run:
Step #19 - "compile-afl-address-x86_64": python infra/helper.py build_image gstreamer
Step #19 - "compile-afl-address-x86_64": python infra/helper.py build_fuzzers --sanitizer address --engine afl --architecture x86_64 gstreamer
```

`git`:
```
Step #17: HEAD is now at 32076d13b7 Merge branch 'ed/fsmonitor-on-networked-macos' into next
Step #17: Removing intermediate container d5f6d074f67c
Step #17:  ---> f23a17992967
Step #17: Step 6/12 : RUN mkdir -p oss-fuzz
Step #17:  ---> Running in d502b6cd8900
Step #17: Removing intermediate container d502b6cd8900
Step #17:  ---> 1879850c6d29
Step #17: Step 7/12 : RUN cp fuzz-* ./oss-fuzz/
Step #17:  ---> Running in 6a12d521af9c
Step #17: cp: cannot stat 'fuzz-*': No such file or directory
Step #17: The command '/bin/sh -c cp fuzz-* ./oss-fuzz/' returned a non-zero code: 1
Finished Step #17
ERROR
ERROR: build step 17 "gcr.io/cloud-builders/docker" failed: step exited with non-zero status: 1
```

**Comment by Navidem** [comment]
/gcbrun trial_build.py all --sanitizer address --fuzzing-engine afl

**Comment by Navidem** [comment]
This is still failing: `trial_build.py all --sanitizer address --fuzzing-engine afl`

`git`:
```
Step #17: HEAD is now at 32076d13b7 Merge branch 'ed/fsmonitor-on-networked-macos' into next
Step #17: Removing intermediate container 66dab5e18fbf
Step #17:  ---> 48a1cc7758df
Step #17: Step 6/12 : RUN mkdir -p oss-fuzz
Step #17:  ---> Running in 01d885066099
Step #17: Removing intermediate container 01d885066099
Step #17:  ---> 14bbfedd8b77
Step #17: Step 7/12 : RUN cp fuzz-* ./oss-fuzz/
Step #17:  ---> Running in cfc2f38a957b
Step #17: cp: cannot stat 'fuzz-*': No such file or directory
Step #17: The command '/bin/sh -c cp fuzz-* ./oss-fuzz/' returned a non-zero code: 1
Finished Step #17
ERROR
```

`apache-httpd`:
```
Step #17: Step 3/9 : RUN svn co svn://vcs.exim.org/pcre2/code/trunk pcre2 &&     cd pcre2 &&     ./autogen.sh &&     ./configure &&     make &&     make install
Step #17:  ---> Running in b23436f9afa3
Step #17: svn: E170013: Unable to connect to a repository at URL 'svn://vcs.exim.org/pcre2/code/trunk'
Step #17: svn: E000111: Can't connect to host 'vcs.exim.org': Connection refused
Step #17: The command '/bin/sh -c svn co svn://vcs.exim.org/pcre2/code/trunk pcre2 &&     cd pcre2 &&     ./autogen.sh &&     ./configure &&     make &&     make install' returned a non-zero code: 1
Finished Step #17
ERROR
```

**Comment by jonathanmetzman** [comment]
> This is still failing: `trial_build.py all --sanitizer address --fuzzing-engine afl`
> 
> `git`:
> 
> ```
> Step #17: HEAD is now at 32076d13b7 Merge branch 'ed/fsmonitor-on-networked-macos' into next
> Step #17: Removing intermediate container 66dab5e18fbf
> Step #17:  ---> 48a1cc7758df
> Step #17: Step 6/12 : RUN mkdir -p oss-fuzz
> Step #17:  ---> Running in 01d885066099
> Step #17: Removing intermediate container 01d885066099
> Step #17:  ---> 14bbfedd8b77
> Step #17: Step 7/12 : RUN cp fuzz-* ./oss-fuzz/
> Step #17:  ---> Running in cfc2f38a957b
> Step #17: cp: cannot stat 'fuzz-*': No such file or directory
> Step #17: The command '/bin/sh -c cp fuzz-* ./oss-fuzz/' returned a non-zero code: 1
> Finished Step #17
> ERROR
> ```
> 
> `apache-httpd`:
> 
> ```
> Step #17: Step 3/9 : RUN svn co svn://vcs.exim.org/pcre2/code/trunk pcre2 &&     cd pcre2 &&     ./autogen.sh &&     ./configure &&     make &&     make install
> Step #17:  ---> Running in b23436f9afa3
> Step #17: svn: E170013: Unable to connect to a repository at URL 'svn://vcs.exim.org/pcre2/code/trunk'
> Step #17: svn: E000111: Can't connect to host 'vcs.exim.org': Connection refused
> Step #17: The command '/bin/sh -c svn co svn://vcs.exim.org/pcre2/code/trunk pcre2 &&     cd pcre2 &&     ./autogen.sh &&     ./configure &&     make &&     make install' returned a non-zero code: 1
> Finished Step #17
> ERROR
> ```

What's your opinion on these failures? Do you think they are related to your change?
before you submit this let's make sure to do a trial build with all sanitizers (including coverage) and all engines and all projects.

**Comment by jonathanmetzman** [comment]
@DavidKorczynski @AdamKorcz do you see any problem in principle in just using a fuzz introspector-compatible compiler in all of our builds so we don't need the clunky `:introspector` images

**Comment by DavidKorczynski** [comment]
> do you see any problem in principle in just using a fuzz introspector-compatible compiler in all of our builds

No, I do not see any problem.

The reasons being: 
Fuzz Introspector's code will only be invoked under LTO (because we patch it into the LTO compilation pipeline) and, if LTO is enabled (e.g. if a build uses lto by default) then the first fuzz introspector checks is whether `FUZZ_INTROSPECTOR` environment variable is set and simply returns from llvm pass if it's not:
https://github.com/ossf/fuzz-introspector/blob/914111b4e6ae5bd7c0f43a648b0a689c2d516122/frontends/llvm/lib/Transforms/FuzzIntrospector/FuzzIntrospector.cpp#L391-L396

It has the impact fuzz introspector should remain compatible with the clang desired -- but this is only a minor issue I recon, in that we already desire to do so.

**Comment by jonathanmetzman** [comment]
Now that the issues you've seen have been fixed in https://github.com/google/oss-fuzz/commit/871ea5240a4cadbc9fd9a78cf5b1e258734fe6cf can you try landing this again?

**Comment by Navidem** [comment]
trial_build.py all --sanitizer address --fuzzing-engine afl honggfuzz centipede

**Comment by Navidem** [comment]
/gcbrun trial_build.py all --sanitizer address --fuzzing-engine afl honggfuzz centipede

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py all --sanitizer address coverage --fuzzing-engine libfuzzer afl honggfuzz centipede

**Comment by Navidem** [comment]
This timed out: `trial_build.py all --sanitizer address coverage --fuzzing-engine libfuzzer afl honggfuzz centipede`

**Comment by Navidem** [comment]
/gcbrun trial_build.py all --sanitizer coverage --fuzzing-engine libfuzzer

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py all --sanitizer coverage address memory undefined introspector --fuzzing-engine libfuzzer honggfuzz afl centipede

**Comment by jonathanmetzman** [comment]
Navid I think this is safe to land, just fix the presubmit issue.

**Comment by Navidem** [comment]
> Navid I think this is safe to land, just fix the presubmit issue (install from requirements.txt and then python infra/presubmit.py format)

Thanks!

File: infra/base-images/base-clang/Dockerfile [hunk 1]
```diff
@@ -23,7 +23,7 @@ ARG arch=x86_64
 ARG introspector
 ENV INTROSPECTOR_PATCHES=$introspector
 ENV FUZZINTRO_OUTDIR=$SRC
-ENV FUZZ_INTROSPECTOR=$introspector
```
Comment (jonathanmetzman) [inline infra/base-images/base-clang/Dockerfile:26]: How will FUZZ_INTROSPECTOR be set anymore for checkout_and_build_clang to be used?

**Review by jonathanmetzman — COMMENTED** [review]
Also, I think all introspector builds are broken now.
Trial builds also seems to have broken yesterday but I think before this landed somehow. Not sure they are related

**Pull request merged**